### PR TITLE
DWGUGens: fix float/double confusion for GCC 6 compatibility (take 2)

### LINKED ISSUE #106: source/DWGUGens/dwglib/DWG.hpp fail to build with GCC 6
The source fail to build in Debian after Debian switched to GCC 6 as the default C++ compiler.  See https://bugs.debian.org/831078 for details.

The problem is that this expression no longer work in source/DWGUGens/dwglib/DWG.hpp:
```
float loss;
...
void set_area(float a1,float lossF)
    {
        if(a1 < 1e-18)
            loss = 0.0;
        else
            loss = 1.0 - std::min(lossF/sqrt(static_cast<double>(a1)),1.0);

    }
```

I believe the cause is that the compiler now pick the function float sqrt(float) instead of the function double sqrt(double).  There are two ways to solve the build problem, and I am unsure which one is the best.  Either replace 1.0 with 1.0f to use float everywhere in the calculations, or cast a1 using static_cast(double>(a1) to tell the compiler to use doubles everywhere.  I assume the former calculation is slightly quicker, while the latter is more accurate.

Do you have an opinion on how this should be solved?  How important is the accuracy of the calculation?

### PR DESCRIPTION
**Pull request by snappizz**
Fix #106.

### CODE REVIEW

**Comment by sonoro1234** [comment]
With sc_min instead of std::min this seems not neccessary: see #106

**Comment by bagong** [comment]
Can't test higher than gcc4.8, but no prob on Windows.
